src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -151,6 +157,38 @@ var ReactDOMInput = {
 
     if (__DEV__) {
       warnIfValueIsNull(props);
+
+      var initialValue = inst._wrapperState.initialChecked || inst._wrapperState.initialValue;
+      var defaultValue = props.defaultChecked || props.defaultValue;
+      var controlledValue = props.checked || props.value;
```

Comment:
This can't be right I think? `false || undefined => undefined` and then later `controlledValue !== undefined`. Why not compute `controlled` instead of value? The value isn't used anywhere.